lib/database_connection.rb

```diff
@@ -0,0 +1,15 @@
+require 'pg'
+
+class DatabaseConnection
+  def self.setup(dbname)
+    @connection = PG.connect(dbname: dbname)
+  end
+
+  class << self
+    attr_reader :connection
+  end
```

Comment:
I've never seen the attr_reader in a method before, would be good to understand the advantages of this syntax!